projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad-task-definitions.php

```diff
@@ -1097,6 +1110,20 @@ function wpcom_launchpad_is_videopress_launch_disabled() {
 	return ! wpcom_is_checklist_task_complete( 'videopress_upload' );
 }
 
+/**
+ * Determines whether or not the blog launch task is enabled
+ *
+ * @return boolean True if blog launch task is enabled
+ */
+function wpcom_launchpad_is_blog_launched_disabled() {
+	if ( 'design-first' === get_option( 'site_intent' ) ) {
+		return ! wpcom_is_checklist_task_complete( 'plan_completed' )
+			|| ! wpcom_is_checklist_task_complete( 'domain_upsell' )
+			|| ! wpcom_is_checklist_task_complete( 'setup_blog' );
```

Comment:
This condition ends up being relatively complex with the three clauses that are all negated and ORed together. It might be a bit clearer if we flip this around a bit.
```suggestion
		// We only want the blog_launched task enabled after other key tasks are all complete.
		if ( wpcom_is_checklist_task_complete( 'plan_completed' )
			&& wpcom_is_checklist_task_complete( 'domain_upsell' )
			&& wpcom_is_checklist_task_complete( 'setup_blog' ) )
		{
			return true;
		}
		return false;
```